typing/env.ml

```diff
@@ -122,6 +122,9 @@ module TycompTbl =
           bindings for each name, as in comp_labels and
           comp_constrs. *)
 
+      root: Path.t;
```

Comment:
I think we can skip adding this field since you are already checking the path when you find it in the summary.

Reply:
I persist to think that this approach is inherently more fragile, so I'm not going to throw away security nets :-)